docs/dev/hacking-guide.md

```diff
@@ -0,0 +1,177 @@
+
+```bash
+oc scale deployment/cluster-cloud-controller-manager -n openshift-cloud-controller-manager-operator --replicas=0
+```
+
+Finally, once all has been scaled down you can compile and run the controller.
+
+You can copy or edit `hack/example-images.json` with the provider image you want to run.
+
+```bash
+make build
+ ./bin/cluster-cloud-controller-manager --images-json=hack/example-images.json
+```
+
+## How to build the operator in a container for remote testing
+
+Prerequisites:
+
+```bash
+git checkout github.com/openshit/$repository_name
+cd $repository_name
+```
+
+First you’ll need to build an image with your changes.
+
+```bash
+IMG="quay.io/your-repo/cluster-cloud-controller-manager-operator:latest" make images
+docker push quay.io/your-repo/cluster-cloud-controller-manager-operator:latest
+```
+
+Second step, similar to running the component locally we need to scale down CVO.
+
+```bash
+oc scale --replicas=0 deployment/cluster-version-operator -n openshift-cluster-version
+```
+
+Next, edit machine-api-operator-images ConfigMap and place a link to your image there.
+
+The clsuter-cloud-controller-manager-operator project contains all the resources and components that are used by the CCCMO. There is a ConfigMap named cloud-controller-manager-images. This ConfigMap contains references to all the images used by the CCCMO, it uses these to deploy the CCM and ensure they are the proper images. The ConfigMap looks something like this:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cloud-controller-manager-images
+  namespace: openshift-cloud-controller-manager-operator
+data:
+  images.json: |
+    {
+      "cloudControllerManagerAWS": "registry.ci.openshift.org/openshift:aws-cloud-controller-manager",
+      "cloudControllerManagerOpenStack": "registry.ci.openshift.org/openshift:openstack-cloud-controller-manager"
+    }
+```
+
+```bash
+oc edit configmap/cloud-controller-manager-images -n openshift-cloud-controller-manager-operator
+```
+
+With the new image information loaded into the ConfigMap, the next thing you might do is replace the CCCMO. This operator controls how the specific cloud controllers are deployed and coordinated. You only change this component if there is something you are testing.
+
+The easiest way to change this operator is to change the image reference in the deployment. The commands you can use:
+
+```bash
+oc edit deployment/cluster-cloud-controller-manager -n openshift-cloud-controller-manager-operator
+```
+
+After changing the download reference in the images ConfigMap an easy way to swap out the controller is to let the CCCMO do it for you. You can delete the deployment associated with the cloud provider CCM and then the CCCMO will create a new one for you, like this:
+
+```bash
+oc delete deployment/aws-cloud-controller-manager -n openshift-cloud-controller-manager
+```
```

Comment:
I'm a bit confused, can we add more details to the beginning of the sections in this doc to explain what we are achieving here, I thought we were hacking on CCCMO but this suggests we are hacking on CCM itself?

Reply:
It is a dev guide. Hacking is a nature of it, and it makes sense to name it so, as longer term this would become a nice-to-have dev documentation for all and everything.

Reply:
imo, seems like both activities are inter-linked. i think as we grow this into the devel style doc that Danil is talking about it will be come more important to create separate CCM and CCCMO sections, but for now this is reading ok to me.